crowbar_framework/app/constraints/api_constraint.rb

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright 2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class ApiConstraint
+  attr_reader :versions
+
+  def initialize(*versions)
+    @versions = versions.map { |v| v.to_s.split(".").map(&:to_i) }
+  end
+
+  def matches?(request)
+    versions.any? do |major, minor|
```

Comment:
> OK, that might be just me and my mental model.
> Sorry for starting this thread, it's completely irrelevant to the PR at question.

@kirushik no problem, I got the point though, which was to cover all areas of the code if possible.

are you OK with the changes now, or is there anything critical you would do differently?